### PR TITLE
Compaction reshard task

### PR DESCRIPTION
**Pull request by Deexie**
Task manager tasks covering reshard compaction.

Reattempt on https://github.com/scylladb/scylladb/pull/14044. Bugfix for https://github.com/scylladb/scylladb/issues/14618 is squashed with 95191f4. 
Regression test added.

### CODE REVIEW

**Comment by Deexie** [comment]
- rebase

File: compaction/task_manager_module.cc [hunk 1]
```diff
@@ -549,7 +522,29 @@ future<> shard_reshaping_compaction_task_impl::run() {
...
     auto destinations = co_await distribute_reshard_jobs(std::move(all_jobs));
-    co_await run_resharding_jobs(_dir, std::move(destinations), _db, _status.keyspace, _status.table, std::move(_creator), std::move(_owned_ranges_ptr));
+
+    uint64_t total_size = boost::accumulate(destinations | boost::adaptors::transformed(std::mem_fn(&replica::reshard_shard_descriptor::size)), uint64_t(0));
+    if (total_size == 0) {
+        co_return;
+    }
+
+    auto start = std::chrono::steady_clock::now();
+    dblog.info("Resharding {} for {}.{}", utils::pretty_printed_data_size(total_size), _status.keyspace, _status.table);
```
Comment (bhalevy) [inline compaction/task_manager_module.cc:532]: should it still use dblog?
not the compaction logger?

File: compaction/task_manager_module.cc [hunk 2]
```diff
@@ -6,13 +6,200 @@
...
+                             sharded<replica::database>& db, sstring ks_name, sstring table_name, sstables::compaction_sstable_creator_fn creator,
+                             compaction::owned_ranges_ptr owned_ranges_ptr) {
+
+    uint64_t total_size = boost::accumulate(reshard_jobs | boost::adaptors::transformed(std::mem_fn(&replica::reshard_shard_descriptor::size)), uint64_t(0));
+    if (total_size == 0) {
+        co_return;
+    }
+
+    auto start = std::chrono::steady_clock::now();
+    dblog.info("Resharding {} for {}.{}", utils::pretty_printed_data_size(total_size), ks_name, table_name);
```
Comment (bhalevy) [inline compaction/task_manager_module.cc]: Not sure that dblog should be used here, or the compaction related logger.

**Review by bhalevy — COMMENTED** [review]
Looks good overall, I had just minor comments

**Comment by Deexie** [comment]
- squash copying, adding, and deleteing to one commit

File: compaction/task_manager_module.cc (hunk 1 above)
Comment (bhalevy) [reply compaction/task_manager_module.cc:532]: We discussed that offline and decided to keep it this way for the time being. This is consistent with tasks previously moved here already.

**Review by denesb — APPROVED** [review]

**Pull request merged**